dali/python/nvidia/dali/_multiproc/pool.py

```diff
@@ -149,11 +148,16 @@ def is_not_received(self, batch_i, tasks):
         """Check if we didn't receive all results for given tasks and batch_i"""
         return len(self.partially_received[batch_i]) < len(tasks)
 
+    def clear_scheduled(self, batch_i):
```

Comment:
This is the fix?

Reply:
I'm afraid that there is a lot of unrelated to this fix changes in this PR. How about extracting only the fix and test for it to a small PR and move the rest of the refactor to a separate one.

Reply:
I removed the unrelated test case from the test_pool. There's one more unrelated change - I dropped two test cases that were checking that the buffer is overwritten if we schedule too much, because I believe it doesn't test anything in fact. Should I bring those two back?